### PR TITLE
Vesting tests cleanup

### PR DESCRIPTION
**Pull request by guzzit**
This PR does a bit of clean up in pallet-vesting with the addition of an add_vest function to abstract some vested transfers.

Resolves [Clean up pallet-vesting tests #9345](https://github.com/paritytech/substrate/issues/9345)

### CODE REVIEW

File: frame/vesting/src/mock.rs [hunk 1]
```diff
@@ -150,7 +151,23 @@ impl ExtBuilder {
 			.assimilate_storage(&mut t)
 			.unwrap();
 		let mut ext = sp_io::TestExternalities::new(t);
-		ext.execute_with(|| System::set_block_number(1));
+		ext.execute_with(|| {
+            System::set_block_number(1);
+            self.schedules.iter().for_each(|schedule| {
```
Comment (ggwpez) [inline frame/vesting/src/mock.rs:156]:
```suggestion
            self.schedules.into_iter().for_each(|(who, schedule)| {
```

File: frame/vesting/src/tests.rs [hunk 2]
```diff
@@ -96,50 +96,41 @@ fn check_vesting_status() {
...
+	ExtBuilder::default().existential_deposit(ED).add_vest(2, ED * 10, ED, 0)
+	.add_vest(2, ED * 30, ED, 5).build().execute_with(|| {
 		assert_eq!(System::block_number(), 1);
 		let sched0 = VestingInfo::new(
 			ED * 20,
 			ED, // Vesting over 20 blocks
 			10,
 		);
-		// Account 2 already has a vesting schedule.
-		assert_eq!(Vesting::vesting(&2).unwrap(), vec![sched0]);
```
Comment (ggwpez) [inline frame/vesting/src/tests.rs:109]: Why did you delete these assertions? Please rather fix them instead of deleting.

File: frame/vesting/src/mock.rs [hunk 3]
```diff
@@ -101,11 +101,12 @@ impl Config for Test {
 pub struct ExtBuilder {
 	existential_deposit: u64,
 	vesting_genesis_config: Option<Vec<(u64, u64, u64, u64)>>,
+	schedules: Vec<(u64, VestingInfo<u64, u64>)>,
 }
 
 impl Default for ExtBuilder {
 	fn default() -> Self {
-		Self { existential_deposit: 1, vesting_genesis_config: None }
+		Self { existential_deposit: 1, vesting_genesis_config: None, schedules: vec![] }
```
Comment (ggwpez) [inline frame/vesting/src/mock.rs:109]: Why did you add another field? Can you not just put it into `vesting_genesis_config`?  
I would rather have one field, so you can either put it all in `vesting_genesis_config` or in `schedules` which is probably cleaner since it has a type instead of a tuple.

**Review by ggwpez — CHANGES_REQUESTED** [review]
Thanks for looking into this!

File: frame/vesting/src/mock.rs [hunk 4]
```diff
@@ -150,7 +151,23 @@ impl ExtBuilder {
 			.assimilate_storage(&mut t)
 			.unwrap();
 		let mut ext = sp_io::TestExternalities::new(t);
-		ext.execute_with(|| System::set_block_number(1));
+		ext.execute_with(|| {
+            System::set_block_number(1);
+            self.schedules.into_iter().for_each(|(who, schedule)| {
+                Vesting::vested_transfer(Some(13).into(), who, schedule).unwrap();
```
Comment (bkchr) [inline frame/vesting/src/mock.rs:157]: The `Some(13)` here needs to be configurable.

File: frame/vesting/src/tests.rs [hunk 5]
```diff
@@ -96,50 +96,41 @@ fn check_vesting_status() {
 	});
 }
 
-#[test]
+#[test]	
 fn check_vesting_status_for_multi_schedule_account() {
-	ExtBuilder::default().existential_deposit(ED).build().execute_with(|| {
+	ExtBuilder::default().existential_deposit(ED).add_vest(2, ED * 10, ED, 0)
```
Comment (bkchr) [inline frame/vesting/src/tests.rs:101]: Misses a schedule.

**Review by bkchr — COMMENTED** [review]
You are missing vesting schedules in your changes. You also did not make the `add_vest` call generic enough.

File: frame/vesting/src/mock.rs (hunk 3 above)
Comment (guzzit) [reply frame/vesting/src/mock.rs:109]: I don't think I can put them into `vesting_genesis_config`. When I did some tests checking `usable_balance` on accounts with multiple schedules failed, since the genesis build function would only update the vesting lock with the details of the last schedule of that account in the iteration.

Then when I tried to make them all schedules instead and call `vested_transfer` on them, the `generates_multiple_schedules_from_genesis_config` and `multiple_schedules_from_genesis_config_errors` tests fail.

File: frame/vesting/src/tests.rs [hunk 6]
```diff
@@ -96,50 +96,41 @@ fn check_vesting_status() {
...
 		let sched2 = VestingInfo::new(
 			ED * 30,
 			ED, // Vesting over 30 blocks
 			5,
 		);
-		assert_ok!(Vesting::vested_transfer(Some(4).into(), 2, sched2));
 
-		System::set_block_number(9);
+		// Account 2 already has 3 vesting schedules.
+		assert_eq!(Vesting::vesting(&2).unwrap(), vec![sched0, sched1, sched2]);
```
Comment (guzzit) [inline frame/vesting/src/tests.rs:123]: Hi @ggwpez, I had added extra assertions here. Or do you mean I should do something like:

```
// Account 2 has more than 1 schedule.
assert_ne!(Vesting::vesting(&2).unwrap(), vec![sched0]);
```

File: frame/vesting/src/tests.rs (hunk 5 above)
Comment (guzzit) [reply frame/vesting/src/tests.rs:101]: I believe the missing schedules are added by default in the genesis config by the mock file. Should I use `add_vest` to add them instead?
